lucene/lucene-directory/src/main/java/org/infinispan/lucene/directory/DirectoryBuilder.java

```diff
@@ -42,8 +43,12 @@ public static BuildContext newDirectoryInstance(Cache<?, ?> metadataCache, Cache
     * Search
     */
    private static Cache<?, ?> decorateCache(Cache<?, ?> cache) {
-      if (cache == null) return null;
-      return cache.getAdvancedCache().withStorageMediaType();
+      if(cache == null) return null;
+      if(cache.getAdvancedCache().getValueDataConversion().getStorageMediaType().equals(MediaType.APPLICATION_UNKNOWN)) {
```

Comment:
What's the expected media type for a Lucene cache? Couldn't you test for that instead of `!APPLICATION_UNKNOWN`?

Reply:
The user can create/configure Lucene caches, so there is no way of knowing. In any case, 
Lucene caches are being removed on https://github.com/infinispan/infinispan/pull/8137.

Reply:
I meant e.g. `org.infinispan.query.impl.LifecycleManager#getDefaultLuceneCacheConfig` uses `MediaType.APPLICATION_OBJECT_TYPE`, shouldn't you say that the cache is custom if the media type is != APPLICATION_OBJECT_TYPE?

In practice the result may be the same, but the comment isn't very clear to me.

Reply:
if I disable transcoding when ```mediaType != APPLICATION_OBJECT_TYPE``` it will break Lucene caches that are configured with OFF_HEAP/BINARY.
But Lucene caches are literally hours ahead of being removed from master, so I will not try changing this :)

Reply:
Ok :)